src/app/app.component.html

```diff
@@ -1,73 +1,88 @@
-       routerLinkActive="active" skipLocationChange="true">
-        <mat-icon>grid_on</mat-icon>
-    </a>
+  <a *ngIf="!(_config | GetConfigVariablePipe:competitionHost)" mat-icon-button matTooltip="Gallery view"
+     routerLink="/gallery"
+     routerLinkActive="active" skipLocationChange="true">
+    <mat-icon>grid_on</mat-icon>
+  </a>
 
-    <span class="spacer-medium"></span>
+  <span class="spacer-medium"></span>
 
-    <a mat-icon-button matTooltip="Mini-gallery view" routerLink="/mini-gallery" routerLinkActive="active"
-       skipLocationChange="true" (click)="setActiveView(0)" [color]="_active_view === 0 ? 'accent' : ''">
-        <mat-icon>view_comfy</mat-icon>
-    </a>
+  <a mat-icon-button matTooltip="Mini-gallery view" routerLink="/mini-gallery" routerLinkActive="active"
+     skipLocationChange="true" (click)="setActiveView(0)" [color]="_active_view === 0 ? 'accent' : ''">
+    <mat-icon>view_comfy</mat-icon>
+  </a>
 
-    <span class="spacer-medium"></span>
+  <span class="spacer-medium"></span>
 
-    <a mat-icon-button matTooltip="List view" routerLink="/list" routerLinkActive="active" skipLocationChange="true"
-       (click)="setActiveView(1)" [color]="_active_view === 1 ? 'accent' : ''">
-        <mat-icon>list</mat-icon>
-    </a>
+  <a mat-icon-button matTooltip="List view" routerLink="/list" routerLinkActive="active" skipLocationChange="true"
+     (click)="setActiveView(1)" [color]="_active_view === 1 ? 'accent' : ''">
+    <mat-icon>list</mat-icon>
+  </a>
 
-    <span class="spacer-medium"></span>
+  <span class="spacer-medium"></span>
 
-    <a mat-icon-button matTooltip="Temporal Scoring view" routerLink="/temporal-list" routerLinkActive="active"
-       skipLocationChange="true" (click)="setActiveView(2)" [color]="_active_view === 2 ? 'accent' : ''">
-        <mat-icon>filter_list</mat-icon>
-    </a>
+  <a mat-icon-button matTooltip="Temporal Scoring view" routerLink="/temporal-list" routerLinkActive="active"
+     skipLocationChange="true" (click)="setActiveView(2)" [color]="_active_view === 2 ? 'accent' : ''">
+    <mat-icon>filter_list</mat-icon>
+  </a>
 
-    <span class="spacer-xlarge"></span>
+  <span class="spacer-xlarge"></span>
 
-    <button (click)="showHistory()" mat-icon-button matTooltip="Show query history.">
-        <mat-icon>history</mat-icon>
-    </button>
+  <button (click)="showHistory()" mat-icon-button matTooltip="Show query history.">
+    <mat-icon>history</mat-icon>
+  </button>
 
-    <button (click)="settingssidenav.toggle()" mat-icon-button [matBadge]="settingsbadge"
-            matTooltip="Toggle query refinement sidebar">
-        <mat-icon>storage</mat-icon>
-    </button>
+  <button (click)="settingssidenav.toggle()" mat-icon-button [matBadge]="settingsbadge"
+          matTooltip="Toggle query refinement sidebar">
+    <mat-icon>storage</mat-icon>
+  </button>
 
+  <ng-container *ngIf="isCompetitionActive()">
```

Comment:
This should pipe the config into an actual angular pipe, otherwise this call is performed every time the mouse is moved, see e.g. https://github.com/vitrivr/vitrivr-ng/blob/master/src/app/query/containers/query-container.component.html

Reply:
This has been addressed. For future reference (I did only touch the `CompetitionEnabledPipe`): I'm not sure whether this is clever or a plain abuse of pipes. What I do know for sure is, by the [angular style guide](https://angular.io/guide/styleguide), pipes should be camelCase